src/syntax/FStar.Syntax.Free.fs

```diff
@@ -77,7 +82,7 @@ let rec free_univs u = match Subst.compress_univ u with
 //to handle that, use_cache flag is UNSET when asking for free_fvars, so that all the terms are traversed completely
 //on the other hand, for earlier interface use_cache is true
 //this flag is propagated, and is used in the function should_invalidate_cache below
-let rec free_names_and_uvs' tm use_cache : free_vars_and_fvars =
+let rec free_names_and_uvs' tm (use_cache:use_cache_t) : free_vars_and_fvars =
```

Comment:
I added a control knob to free_names_and_uvs to enable it to compute free_uvars of a Tm_uvar while also including the free uars of ctx_uvar_typ. It's useful for future experimentation, but in the current version, this argument is never set to Full